### ISSUE TITLE
Add delete button to MDCChipView

### ISSUE THREAD

**Issue by tbechtum** [material-components/material-components-ios#7758]
## Feature Request/Suggestion

Would be nice to have the MDCChipField.addClearButtonToChip/drawClearButton functions also available in the MDCChipView implementation in order to achieve this filter collection view

![Screenshot 2019-07-01 at 09 49 28](https://user-images.githubusercontent.com/8265121/60422479-69bb9900-9bec-11e9-9c44-530ca3b19c77.png).

## Additional context

This is my trial code for the above screen shot

1. Exported MDCChipField.addClearButtonToChip

`- (nonnull UIImage *)drawClearButton;`

2. Collection cell configuration

```
    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: reuseIdentifier, for: indexPath) as! MDCChipCollectionViewCell
        
        cell.chipView.mdc_adjustsFontForContentSizeCategory = true
        cell.alwaysAnimateResize = true
        cell.chipView.titleLabel.text = self.titles[indexPath.row]
        cell.chipView.setTitleColor(UIColor.settings.uni
            , for: .normal)
        cell.chipView.titleLabel.textColor = UIColor.settings.uni
        cell.chipView.enableRippleBehavior = true
        cell.chipView.selectedImageView.image = nil // possible identifying image when selecting
        cell.chipView.rippleAllowsSelection = false 
        // KNOWN ISSUE: Google's chip view for iOS (MDCChipView) does not have the the ability to add the required delete button, but the other component (MDCChipField) has this ability.

        let closeActionImageView = UIImageView(image: MDCChipField().drawClearButton())
        cell.chipView.accessoryView = closeActionImageView
        return cell
    }

```

**Comment by anuragfluke** [comment]
is there any ETA for this bug ?

**Comment by romoore** [comment]
I do not believe this Issue is currently planned for active work.  The primary challenge is that MDCChipView does not easily meet minimum guidelines for touch targets (48 x 48 points) and adding a delete button with its own 48x48 point touch target can make the behavior even less accessible.   If you want to add a clear button, you can add a UIButton with a clear icon to the `accessoryView` of the Chip and attach target/action  handlers for the button.

**Comment by phamduc0810** [comment]
> I do not believe this Issue is currently planned for active work. The primary challenge is that MDCChipView does not easily meet minimum guidelines for touch targets (48 x 48 points) and adding a delete button with its own 48x48 point touch target can make the behavior even less accessible. If you want to add a clear button, you can add a UIButton with a clear icon to the `accessoryView` of the Chip and attach target/action handlers for the button.

Could you show me the code, I have tried but still not success.
This is my try:
`my code:

        let chipField = MDCChipField()
        chipField.showChipsDeleteButton = true
        chipField.translatesAutoresizingMaskIntoConstraints = false
        chipField.minTextFieldWidth = self.view.frame.width
        var tiltle = "a"
        for _ in 1...20 {
            let chipView = MDCChipView()
            tiltle += "a"
            chipView.titleLabel.text = tiltle
            chipView.setTitleColor(UIColor.red, for: .selected)
            chipView.sizeToFit()
            chipView.accessoryView?.backgroundColor = .black
            let button = UIButton()
            button.titleLabel?.text = "button"
            chipView.accessoryView?.addSubview(button)
            chipView.addTarget(self, action: #selector(tapChipView), for: .touchUpInside)
            chipField.addChip(chipView)
        }
        chipField.textField.isUserInteractionEnabled = false
        scrollView.addSubview(chipField)`

**Comment by romoore** [comment]
Sorry if I was unclear in my last comment. You should assign the button as the `accessoryView` of the Chip rather than adding a button to the accessory view.

**Comment by phamduc0810** [comment]
> Oops, I had a mix-up in my last comment. You should assign the button as the `accessoryView` of the Chip rather than adding a button to the accessory view. Sorry for any confusion I caused.

thanks, it's worked.
